src/components/Menu/MenuItem.tsx

```diff
@@ -121,51 +129,71 @@ class MenuItem extends UIComponent<Extendable<IMenuItemProps>, IMenuItemState> {
      */
     renderIcon: PropTypes.func,
 
+    /**
+     * A custom render function the wrapper slot.
+     *
+     * @param {React.ReactType} Component - The computed component for this slot.
+     * @param {object} props - The computed props for this slot.
+     * @param {ReactNode|ReactNodeArray} children - The computed children for this slot.
+     */
+    renderWrapper: PropTypes.func,
+
     /** Additional CSS styles to apply to the component instance.  */
     styles: PropTypes.oneOfType([PropTypes.object, PropTypes.func]),
 
     /** Override for theme site variables to allow modifications of component styling via themes. */
     variables: PropTypes.oneOfType([PropTypes.object, PropTypes.func]),
+
+    /** Shorthand for the wrapper component */
+    wrapper: PropTypes.oneOfType([PropTypes.node, PropTypes.object]),
   }
 
   static defaultProps = {
-    as: 'li',
+    as: 'a',
     accessibility: menuItemBehavior as Accessibility,
+    wrapper: {},
   }
 
   state = IsFromKeyboard.initial
 
   renderComponent({ ElementType, classes, accessibility, rest }) {
-    const { children, content, icon, renderIcon } = this.props
+    const { children, content, icon, renderIcon, renderWrapper, wrapper } = this.props
 
-    return (
+    const menuItemInner = childrenExist(children) ? (
+      children
+    ) : (
       <ElementType
-        className={classes.root}
-        {...accessibility.attributes.root}
-        {...accessibility.keyHandlers.root}
+        className={cx('ui-menu__item__anchor', classes.anchor)}
+        onClick={this.handleClick}
+        onBlur={this.handleBlur}
+        onFocus={this.handleFocus}
+        {...accessibility.attributes.anchor}
+        {...accessibility.keyHandlers.anchor}
         {...rest}
       >
-        {childrenExist(children) ? (
-          children
-        ) : (
-          <a
-            className={cx('ui-menu__item__anchor', classes.anchor)}
-            onClick={this.handleClick}
-            onBlur={this.handleBlur}
-            onFocus={this.handleFocus}
-            {...accessibility.attributes.anchor}
-            {...accessibility.keyHandlers.anchor}
-          >
-            {icon &&
-              Icon.create(this.props.icon, {
-                defaultProps: { xSpacing: !!content ? 'after' : 'none' },
-                render: renderIcon,
-              })}
-            {content}
-          </a>
-        )}
+        {icon &&
+          Icon.create(this.props.icon, {
+            defaultProps: { xSpacing: !!content ? 'after' : 'none' },
+            render: renderIcon,
+          })}
+        {content}
       </ElementType>
     )
+
+    if (wrapper) {
+      return createHTMLListItem(wrapper, {
```

Comment:
don't we want this to be an actual `Slot` shorthand so it's being passed through the Fella renderer? look for `Slot.create` references; we can discuss tomorrow

Reply:
To expand here, we need to pass the content of `wrapper` prop through the Fella renderer in order to leverage Fella's functionality (rtl, styles applied as classnames, etc)

The way to go now is by using `Slot`; I'm doing the same for `Input` in #326 

Right now we need to use `wrapper.style` instead of `wrapper.styles` to style the wrapper and these are applied  directly to `element.style` and not through `classnames`:

![screen shot 2018-10-11 at 13 01 58](https://user-images.githubusercontent.com/5442794/46799793-089a1f80-cd56-11e8-9623-af6719f63156.png)

`wrapper.styles` won't work (it will be added directly to the `DOM` as an attribute), need to use `Slot` for that:

![screen shot 2018-10-11 at 13 02 15](https://user-images.githubusercontent.com/5442794/46799935-69c1f300-cd56-11e8-84b7-c0564c712310.png)